lib/game/widget_builder.dart

```diff
@@ -1,157 +1,210 @@
 import 'package:flutter/widgets.dart';
+import 'package:flutter/gestures.dart';
+
+import './gesture_detector.dart' as flame_detector;
 import '../gestures.dart';
 import 'embedded_game_widget.dart';
 import 'game.dart';
 
+bool _hasGestureDetectors(Game game) =>
+    game is TapDetector ||
+    game is SecondaryTapDetector ||
+    game is DoubleTapDetector ||
+    game is LongPressDetector ||
+    game is VerticalDragDetector ||
+    game is HorizontalDragDetector ||
+    game is ForcePressDetector ||
+    game is PanDetector ||
+    game is ScaleDetector ||
+    game is MultiTouchTapDetector ||
+    game is MultiTouchDragDetector;
+
+Widget _applyGesturesDetectors(Game game, Widget child) {
+  return flame_detector.GestureDetector(
+    // Taps
+    onTap: game is TapDetector ? () => (game as TapDetector).onTap() : null,
```

Comment:
I'm thinking if a function

```
void Function() generateDetector<T>(void Function(T) handler) {
    return game is T ? handler(game as T) : null;
}
```

or something is worth it or to unnecessarily complex? to simplify all these to

```
onTap: generateDetector<TapDetector>((e) => e.onTap()),
```

and etc.

Reply:
well
```
game is T ? () => handler(game as T) : null
```
but you got the idea

Reply:
hmm, not sure if this would work, but I am going to give it a try.

Reply:
@luanpotter I am gonna to leave this suggestion for when we refactor this to solve #199

Reply:
it can def. work, I'm not sure if it's cleaner or not